### PR TITLE
[EHI-53] string exercise added

### PR DESCRIPTION
**Pull request by Ahad726**

### CODE REVIEW

File: src/main/data_structure/string_exercise.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from src.main.logger.py_logger import PyLogger
+
+logger = PyLogger.get_configured_logger()
+
+
+# Count the number of characters (character appearance) in a string
+def char_appearance(string):
```
Comment (jonyroy) [inline src/main/data_structure/string_exercise.py]: change the function name. function name always should be verb

File: src/main/data_structure/string_exercise.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            string += 'ly'
+        else:
+            string += 'ing'
+    return string
+
+
+# Find the first appearance of the substring 'not' and 'poor' from a given string,
+# if 'not' follows the 'poor', replace the whole 'not'...'poor' substring with 'good'.
+# Return the resulting string
+def string_replace(string):
```
Comment (jonyroy) [inline src/main/data_structure/string_exercise.py]: change the function name. function name always should be verb

**Review by jonyroy — CHANGES_REQUESTED** [review]

File: src/main/data_structure/string_exercise.py (hunk 1 above)
Comment (Ahad726) [reply src/main/data_structure/string_exercise.py]: ok

File: src/main/data_structure/string_exercise.py (hunk 2 above)
Comment (Ahad726) [reply src/main/data_structure/string_exercise.py]: ok dada

**Review by jonyroy — APPROVED** [review]

**Pull request merged**